flask_sqlalchemy/__init__.py

```diff
@@ -874,8 +874,14 @@ def apply_driver_hacks(self, app, sa_url, options):
         if sa_url.drivername.startswith('mysql'):
             sa_url.query.setdefault('charset', 'utf8')
             if sa_url.drivername != 'mysql+gaerdbms':
-                options.setdefault('pool_size', 10)
-                options.setdefault('pool_recycle', 7200)
+                try:
+                    pool_class = options['poolclass']
+                except KeyError:
+                    pool_class = None
```

Comment:
Would it be better to do 

`if 'poolclass' in options:`

rather than having a side effect behaviour?

Reply:
I'd define this as defensive programming rather than a side effect, but yes, your suggestion is far better :)

Reply:
Updated

Reply:
I wouldn't go with far better :flushed: 😳

Reply:
What about `pool_class = options.get('poolclass')`?

Reply:
ah nvm just realized the code the comments were on doesn't even exist anymore

Reply:
> What about `pool_class = options.get('poolclass')`?

I like that. A much less expensive operation.
Would it be good to explicitly add a default to make obvious that we selected a `get` to provide a `None`? Although we test for that a couple of lines later - it should be obvious.

Reply:
No, returning `None` is literally the default of `dict.get()` in the stdlib.

Reply:
Same efficiency, but could still reduce the line count from 3 -> 1

Reply:
@ThiefMaster You can tell I have been bitten by idiots in the past 😂

Reply:
I usually expect people to know the basics of the language they are using ;)

Reply:
Too often that's proven to not be the case 😔

Reply:
`options.get` is a nice choice, but you still need to handle None types to retrieve `__name__`, so I don't think there would be any significant benefits over the (now revised) logic.

Reply:
Shouldn't we then be setting the name to `NullType`

Reply:
Ignore that. Being simple.